### DISCUSSION TITLE
Upgrade from 16.1.1 -> 20 - AuthenticatorFactory, DisplayTypeAuthenticatorFactory issue

### DISCUSSION THREAD

**Discussion by jamesonTT** [keycloak/keycloak#22541] [Archive]
Hi, we are using Keycloak 16.1.1 have an SPI which makes use of the `org.keycloak.authentication.AuthenticatorFactory` interface, as part of this we have previously implemented the `org.keycloak.authentication.DisplayTypeAuthenticatorFactory` interface alongside the Authenticator
e.g:

```
public class MyAuthenticatorClass implements AuthenticatorFactory, DisplayTypeAuthenticatorFactory {
... all our stuff
}
```

I'm not able to find reference to this in the javadocs - can anyone help with when it was deprecated and what it was replaced by?

**Comment by Paulal2000** [comment]
I have the same problem. Have you been able to solve it?
